google-ads-examples/src/main/java/com/google/ads/googleads/examples/shoppingads/AddPerformanceMaxRetailCampaign.java

```diff
@@ -658,6 +687,39 @@ private MutateOperation createPerformanceMaxCampaignOperation(
   }
   // [END add_performance_max_retail_campaign_9]
 
+  // [START add_performance_max_retail_campaign_10]
+  /** Creates a list of MutateOperations that create a new asset group listing group filter. */
+  private List<MutateOperation> createAssetGroupListingGroupFilterOperations(
+      String assetGroupResourceName) {
+    List<MutateOperation> mutateOperations = new ArrayList<>();
+
+    // Creates a new asset group listing group filter containing the "default" listing group (All
+    // products).
+    AssetGroupListingGroupFilter listingGroupFilter =
+        AssetGroupListingGroupFilter.newBuilder()
+            .setAssetGroup(assetGroupResourceName)
+            // Does not set the parentListingGroupFilter since this is the root node. For all other
+            // nodes, this would refer to the parent listing group filter resource name.
+            // .setParentListingGroupFilter("<PARENT FILTER RESOURCE NAME>")
+
+            // Sets the type to UNIT_INCLUDED since this node has no children.
+            .setType(ListingGroupFilterType.UNIT_INCLUDED)
+            // Specifies that this is in the shopping vertical, as required for a Performance Max
+            // retail campaign.
+            .setVertical(ListingGroupFilterVertical.SHOPPING)
+            .build();
+
+    // Adds an operation to the list to create the listing group filter.
+    mutateOperations.add(
+        MutateOperation.newBuilder()
+            .setAssetGroupListingGroupFilterOperation(
+                AssetGroupListingGroupFilterOperation.newBuilder().setCreate(listingGroupFilter))
+            .build());
+
+    return mutateOperations;
```

Comment:
We can shorten this to just return the single mutate operation as a list and remove line 694. It seems unnecessary to create a list, add to it, then return the list since there's only one item in the list.